tests/terminate.rs

```diff
@@ -0,0 +1,842 @@
+    e.set_time_sec(GENESIS_TIMESTAMP_SEC + ONE_YEAR_SEC * 2 + ONE_YEAR_SEC * 2 / 3);
+    let lockups = e.get_account_lockups(&users.alice);
+    assert_eq!(lockups[0].1.total_balance, amount * 5 / 8);
+    assert_eq!(lockups[0].1.claimed_balance, amount / 8);
+    assert_eq!(lockups[0].1.unclaimed_balance, amount / 8);
+
+    // claiming
+    let res: WrappedBalance = e.claim(&users.alice).unwrap_json();
+    assert_eq!(res.0, amount / 8);
+
+    // Rewind to 3Y + Y * 2 / 3, 5/8 of original unlock, full vested unlock
+    e.set_time_sec(GENESIS_TIMESTAMP_SEC + ONE_YEAR_SEC * 3 + ONE_YEAR_SEC * 2 / 3);
+    let lockups = e.get_account_lockups(&users.alice);
+    assert_eq!(lockups[0].1.total_balance, amount * 5 / 8);
+    assert_eq!(lockups[0].1.claimed_balance, amount * 2 / 8);
+    assert_eq!(lockups[0].1.unclaimed_balance, amount * 3 / 8);
+
+    // claiming
+    let res: WrappedBalance = e.claim(&users.alice).unwrap_json();
+    assert_eq!(res.0, amount * 3 / 8);
+
+    // Checking lockups
+    let lockups = e.get_account_lockups(&users.alice);
+    assert!(lockups.is_empty());
+
+    // User lockups are empty
+    let lockup = e.get_lockup(lockup_index);
+    assert_eq!(lockup.total_balance, amount * 5 / 8);
+    assert_eq!(lockup.claimed_balance, amount * 5 / 8);
+    assert_eq!(lockup.unclaimed_balance, 0);
+}
+
+#[test]
+fn test_lockup_terminate_custom_vesting_during_lockup_cliff() {
+    let e = Env::init(None);
+    let users = Users::init(&e);
+    let amount = d(60000, TOKEN_DECIMALS);
+    e.set_time_sec(GENESIS_TIMESTAMP_SEC);
+    let lockups = e.get_account_lockups(&users.alice);
+    assert!(lockups.is_empty());
+
+    let (lockup_schedule, vesting_schedule) = lockup_vesting_schedule(amount);
+    let lockup = Lockup {
+        account_id: users.alice.valid_account_id(),
+        schedule: lockup_schedule,
+        claimed_balance: 0,
+        termination_config: Some(TerminationConfig {
+            terminator_id: users.eve.valid_account_id(),
+            vesting_schedule: Some(HashOrSchedule::Schedule(vesting_schedule)),
+        }),
+    };
+
+    let balance: WrappedBalance = e.add_lockup(&e.owner, amount, &lockup).unwrap_json();
+    assert_eq!(balance.0, amount);
+    let lockups = e.get_account_lockups(&users.alice);
+    assert_eq!(lockups.len(), 1);
+    let lockup_index = lockups[0].0;
+
+    // 2Y + Y * 2 / 3, 1/8 unlocked
+    e.set_time_sec(GENESIS_TIMESTAMP_SEC + ONE_YEAR_SEC * 2 + ONE_YEAR_SEC * 2 / 3);
+    let lockups = e.get_account_lockups(&users.alice);
+    assert_eq!(lockups[0].1.total_balance, amount);
+    assert_eq!(lockups[0].1.claimed_balance, 0);
+    assert_eq!(lockups[0].1.unclaimed_balance, amount / 4);
+
+    // Trying to claim
+    ft_storage_deposit(&users.alice, TOKEN_ID, &users.alice.account_id);
+    let res: WrappedBalance = e.claim(&users.alice).unwrap_json();
+    assert_eq!(res.0, amount / 4);
+
+    // TERMINATE, 3Y + Y / 3, 5/6 unlocked
```

Comment:
```suggestion
    // TERMINATE, 3Y + Y / 3, 1/2 unlocked
```

BTW, I find it hard to follow the checkpoints when you jump from `2Y + Y * 2 / 3` to `3Y + Y / 3`. Given 2Y is a cliff, I would define it as cliff and always operate with `cliff + Y * 2 / 3` and `cliff + Y * 4 / 3`

Reply:
Fixed